src/app/shared/components/products/products-list/products-list.component.spec.ts

```diff
@@ -0,0 +1,186 @@
+import {
+  ComponentFixture,
+  fakeAsync,
+  TestBed,
+  tick,
+} from '@angular/core/testing';
+import { ProductsService } from 'src/app/core/products.service';
+import { ProductsListComponent } from './products-list.component';
+import { of } from 'rxjs';
+import { Product } from 'src/app/product.interface';
+import { defer } from 'rxjs';
+import { By } from '@angular/platform-browser';
+import { ProductCardComponent } from '../product-card/product-card.component';
+import { DeleteModalComponent } from '../modals/delete-modal/delete-modal.component';
+import { AddProductModalComponent } from '../modals/add-product-modal/add-product-modal.component';
+import { EditProductModalComponent } from '../modals/edit-product-modal/edit-product-modal.component';
+
+function asyncData<T>(data: T) {
+  return defer(() => Promise.resolve(data));
```

Comment:
Another way to wrap a promise into an observable is with `from`

`from(Promise.resolve())`